modules/vpc-sc/README.md

```diff
@@ -194,6 +211,83 @@ module "test" {
 # tftest modules=1 resources=3 inventory=regular.yaml
 ```
 
+## Factories
+
+This module implements support for three distinct factories, used to create and manage access levels, egress policies and ingress policies via YAML files. The YAML files syntax is a 1:1 match for the corresponding variables, and the factory data is merged at runtime with any data set in variables, which take precedence in case of key overlaps.
```

Comment:
we had a chat on this yesterday. I think the suggestion from Simone is a good compromise.

Otherwise I would mention that file names will work as the key/name of the entity

Reply:
files are parsed together and put in a dictionary

this factory uses single files, as I'm working with a customer with tons of policies: having them all in the same file defeats the purpose, and is equivalent to having a tfvars; single files allow them to quickly find and edit individual policies

Reply:
I fail to understand your point Simo, my doc youc ited above doc mentions

- [x] YAML schema == variable schema: every one of the three resource types has identical schema to the variable objects inside the maps, including optional checks; map keys are the file names, the map itself is the resource folder
- [x] `Factories should consume directories (vs single files)`
- [ ] `All files should contain a dictionary` this is poorly worded in my doc, but if you have a folder with individual files, the folder is the dictionary, the file name is the key
- [x] `Files in a directory should be parsed together and flattened into a single dictionary`

Reply:
I think it's just allowing more than one object per file, like we do for firewall rules

Reply:
I think we should have one or the other, not both. In this case the main benefit is having single files, this is why I'm doing this as I have a customer who needs it.

If you have files as dicts, then you lose naming at the file level 

Project factory and subnets work like this factory.

Reply:
My understanding based on a previous discussion was that we wanted to offer this level of flexibility, at the cost of losing naming at the file level. 
I don't have strong opinions, so you have my vote.

Reply:
But as always you have a point: we should clarify the ADR.